Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -87,7 +88,8 @@ class TabScrollingController: NSObject {
     private var scrollDirection: ScrollDirection = .Down
     private var toolbarState: ToolbarState = .Visible
 
-    override init() {
+    init(profile: Profile) {
```

Comment:
Yup this makes sense! The next step would be to grab the preference key/value and add some conditions in the code below to enable/disable the scrolling.